File: Laboratory/lab5 - PDM a PCM/read_plot_systrace_data.m
% Load the CSV file (assuming no headers)
filename = "systrace_data.csv";

% Read the data using textscan
fid = fopen(filename, 'r');
data = textscan(fid, '%s %f %s %f %s', 'Delimiter', ';');
fclose(fid);

% Extract PCM values and timestamps
pcm_values = data{2};  % Second column contains PCM values
timestamps = 1000*data{4}/168000000;  % Fourth column contains timestamps in ms

% Plot PCM waveform
figure;
plot(timestamps, pcm_values, 'b');
xlabel('Time [ms]');
ylabel('Amplitude [Raw]');
title('PCM Data Captured from SysTrace');
grid on;

File: Laboratory/lab5 - PDM a PCM/read_plot_systrace_data.py
import pandas as pd
import matplotlib.pyplot as plt

# Load the CSV file without headers
filename = "SWV_Data_Trace.txt"

# Read CSV with semicolon as delimiter, ignoring the first and third columns
df = pd.read_csv(filename, delimiter=";", usecols=[1, 3], names=["PCM", "Timestamp"])

# Convert PCM values to integer
y = pd.to_numeric(df["PCM"], errors="coerce")

# Convert timestamps to miliseconds
x = 1000*pd.to_numeric(df["Timestamp"], errors="coerce")/168000000.0

# Plot PCM data over sample index
plt.figure(figsize=(10, 4))
plt.plot(x, y, label="PCM Data")
plt.xlabel("Time [ms]")
plt.ylabel("Amplitude")
plt.title("PCM Data Captured from SysTrace")
plt.legend()
plt.grid()
plt.show()